### ISSUE TITLE
Unnecessary logprob computation in actor.forward

### ISSUE THREAD

**Issue by zkshan2002** [OpenRLHF/OpenRLHF#467]
I believe in `models/actor.py` function `forward`, logprob of prompt logits are not necessary to compute when `not self.packing_samples`.
```
        log_probs = log_probs_from_logits(output["logits"][:, :-1, :], sequences[:, 1:])

        if not self.packing_samples:
            action_log_probs = log_probs[:, -num_actions:]
        else:
            ...
        if return_output:
            return (action_log_probs, output)
        else:
            return action_log_probs
```
This should be equivalent to
```
        action_log_probs = log_probs_from_logits(output["logits"][:, -num_actions:-1, :], sequences[:, -num_actions + 1:])
```
Doing so should save some computation and cuda memory, especially when prompt is long.

**Comment by zkshan2002** [comment]
correction:
```
action_log_probs = log_probs_from_logits(output["logits"][:, :-1, :][:, -num_actions:], sequences[:, 1:][:, -num_actions:])
```
or
```
action_log_probs = log_probs_from_logits(output["logits"][:, -num_actions - 1:-1, :], sequences[:, -num_actions:])
```

**Comment by zkshan2002** [comment]
Hi, is there any update on this issue?
